renode-linux-runner-docker/payload/run-in-renode.py

```diff
@@ -48,6 +50,164 @@ def __getattr__(self, attr):
         return getattr(self.stream, attr)
 
 
+class Action(Protocol):
+    """
+    Called by the add_devices function. Used for some
+    command when additional action is required.
+    """
+
+    """
+    If Action requirements are not satisfied this error will be printed
+    """
+    error: str
+
+    def __call__(self, *args: Any) -> str:
+        raise NotImplementedError
+
+    def correct(self, *args: Any) -> bool:
```

Comment:
`correct` can also be a verb (and method names are usually verb phrases) so it may seem at first glance like this corrects something. A name like `check_args` would fit better, I think.